flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroWriters.java

```diff
@@ -0,0 +1,197 @@
+	 * The Avro writers will use the schema of that specific type to build and write the records.
+	 *
+	 * @param type The class of the type to write.
+	 * @param configurator The configurator to modify the writer properties.
+	 */
+	public static <T extends SpecificRecordBase> AvroWriterFactory<T> forSpecificRecord(
+		Class<T> type,
+		WriterConfigurator<T> configurator) {
+
+		String schemaString = SpecificData.get().getSchema(type).toString();
+		AvroBuilder<T> builder = (out) -> createAvroDataFileWriter(
+			schemaString,
+			AvroRecordType.SPECIFIC,
+			configurator,
+			out);
+		return new AvroWriterFactory<>(builder);
+	}
+
+	/**
+	 * Creates an {@link AvroWriterFactory} that accepts and writes Avro generic types.
+	 * The Avro writers will use the given schema to build and write the records.
+	 *
+	 * @param schema The schema of the generic type.
+	 */
+	public static AvroWriterFactory<GenericRecord> forGenericRecord(Schema schema) {
+		return forGenericRecord(schema, writer -> {});
+	}
+
+	/**
+	 * Creates an {@link AvroWriterFactory} that accepts and writes Avro generic types
+	 * and given <tt>configurator</tt>. The Avro writers will use the given schema to
+	 * build and write the records.
+	 *
+	 * @param schema The schema of the generic type.
+	 * @param configurator The configurator to modify the writer properties.
+	 */
+	public static AvroWriterFactory<GenericRecord> forGenericRecord(
+		Schema schema,
+		WriterConfigurator<GenericRecord> configurator) {
+
+		String schemaString = schema.toString();
+		AvroBuilder<GenericRecord> builder = (out) -> createAvroDataFileWriter(
+			schemaString,
+			AvroRecordType.GENERIC,
+			configurator,
+			out);
+		return new AvroWriterFactory<>(builder);
+	}
+
+	/**
+	 * Creates an {@link AvroWriterFactory} for the given type. The Avro writers will
+	 * use reflection to create the schema for the type and use that schema to write
+	 * the records.
+	 *
+	 * @param type The class of the type to write.
+	 */
+	public static <T> AvroWriterFactory<T> forReflectRecord(Class<T> type) {
+		return forReflectRecord(type, writer -> {});
+	}
+
+	/**
+	 * Creates an {@link AvroWriterFactory} for the given type and given <tt>configurator</tt>.
+	 * The Avro writers will use reflection to create the schema for the type and use that schema
+	 * to write the records.
+	 *
+	 * @param type The class of the type to write.
+	 * @param configurator The configurator to modify the writer properties.
+	 */
+	public static <T> AvroWriterFactory<T> forReflectRecord(
+		Class<T> type,
+		WriterConfigurator<T> configurator) {
+
+		String schemaString = ReflectData.get().getSchema(type).toString();
+		AvroBuilder<T> builder = (out) -> createAvroDataFileWriter(
+			schemaString,
+			AvroRecordType.REFLECT,
+			configurator,
+			out);
+		return new AvroWriterFactory<>(builder);
+	}
+
+	private static <T> DataFileWriter<T> createAvroDataFileWriter(
+		String schemaString,
```

Comment:
We cannot directly pass schema because schema is not serializable, and since shcemaString (or schema) is reference by the anonymous function `builder`, it must be serializable.